src/proof/service/proof-impl.service.ts

```diff
@@ -39,19 +39,43 @@ export class ProofServiceImpl implements ProofService {
       return Promise.reject("Couldn't get proof for specified records")
     }
 
-    return this.verifyProof(proof, network)
+    let finalNetwork: Network
+    if (network) {
+      finalNetwork = network
+    } else {
+      finalNetwork = selectNetwork(proof.anchor.networks[0].name)
```

Comment:
The select network function should take Ethereum Mainnet if included in the network array. If not, then select the first one in the array.